test/crltest.c

```diff
@@ -302,12 +302,11 @@ static STACK_OF(X509_CRL) *make_CRL_stack(X509_CRL *x1, X509_CRL *x2)
 {
     STACK_OF(X509_CRL) *sk = sk_X509_CRL_new_null();
 
-    sk_X509_CRL_push(sk, x1);
-    X509_CRL_up_ref(x1);
-    if (x2 != NULL) {
-        sk_X509_CRL_push(sk, x2);
+    if (x1 != NULL && sk_X509_CRL_push(sk, x1))
+        X509_CRL_up_ref(x1);
+    if (x2 != NULL && sk_X509_CRL_push(sk, x2))
         X509_CRL_up_ref(x2);
-    }
+
```

Comment:
up_ref can fail. Can you please check the return value of up_ref first and only if it succeeds, push it. And free() on failed push of course to down_ref.

Reply:
You're right. It is actually fixed correctly in #26294. I'll just revert this change to avoid the conflict.